bucket.go

```diff
@@ -70,3 +75,95 @@
 	}
 	return nil
 }
+
+// BeginRead starts reading a record from the given entry at the specified timestamp.
+//
+// Parameters:
+//   - ctx: Context for cancellation and timeout control.
+//   - entry: Name of the entry to read from.
+//   - ts: Optional A UNIX timestamp in microseconds. If it is empty, the latest record is returned.
+//   - id: Optional A query ID to read the next record in the query. If it is set, the parameter ts is ignored.
+//   - head: If true, performs a HEAD request to fetch metadata only.
+//
+// It returns a readableRecord or an error if the read fails.
+//
+// Use readableRecord.Read() to read the content of the reader
+func (b *Bucket) BeginRead(ctx context.Context, entry string, ts, id *string, head bool) (*readableRecord, error) {
```

Comment:
the id parameter is used only with the Query endpoints and shouldn't be a part of the public API.

I see that Golang doesn't allow default arguments. Le's create two methods:

BeginRead(...) is used in 95% of use cases
BeginMetadataRead()